File: array/moveZeros.js
// https://leetcode.com/problems/move-zeroes/
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
const moveZeroes = (nums) => {
  let lastZeroRef = 0;
  nums.forEach((value, i) => {
    if (value !== 0) {
      [nums[i], nums[lastZeroRef]] = [nums[lastZeroRef], nums[i]];
      lastZeroRef++;
    }
  });
};

File: array/moveZeros.py
# https://leetcode.com/problems/move-zeroes/
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        zero = 0  # records the position of "0"
        for i in range(len(nums)):
            if nums[i] != 0:
                nums[i], nums[zero] = nums[zero], nums[i]
                zero += 1
        